File: 003-fact0r.py
#!/usr/bin/python3

from math import sqrt
TARGET= 600851475143
#600851475143

nomore=sqrt(TARGET)
primes=[2,3,5,7,11,13]
factors=[]

def nextPrime():
	a=primes[len(primes)-1]+2
	while (not isPrime(a)):
		a+=1
	primes.append(a)
	return a

def isPrime(b):
	for i in primes:
		if (b%i==0):
			return 0
	return 1

def isInteger(x): return int(x) == x



i=nextPrime()
while (i<nomore):
	i=nextPrime()


print("primes: ")
print(len(primes))
print("max: ")
print(primes[len(primes)-1])
print("wynik: ")
while (1):
	j=primes.pop()
	if (TARGET%j==0 and isPrime(TARGET/j)):
		print(j)
		break
		

#	proba=TARGET/j
#	if (isInteger(proba) and isPrime(proba)):
#		print proba

#	if (TARGET%j==0 and isPrime(TARGET/j)):
#		print TARGET/j

File: 003-fact0r.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>


#define TARGET 600851475143LL

// 600851475143

long long* primes;
int countPrimes;

void addPrime(long long newPrime) 
{ 
	primes[countPrimes]=newPrime; 
	countPrimes++;
}

long long lastPrime()
{
	return primes[countPrimes-1];
}

int isPrime(long long b)
{
	int i;
	for(i=0;i<countPrimes;++i)
	{
		int c=primes[i];
		//printf("c=%d ",c);		
		if(b%c==0)
			return 0;
	}
	return 1;
}

long long nextPrime()
{
	long long a=lastPrime()+2;
	while (! isPrime(a))
		a++;
	addPrime(a);
	return a;	
}

int main()
{
	long long biggestPossible=floor(sqrt(TARGET));
	printf("biggestPossible %lld ",biggestPossible);
	primes=malloc(biggestPossible);
	countPrimes=0;
	addPrime(2); addPrime(3);
	long long i=nextPrime();
	while(i<biggestPossible)
	{
		//printf("%d ",i);
		i=nextPrime();
	}
	printf("primes: %d \n",countPrimes);
	printf("max: %lld \n",lastPrime());
	int gdzie=countPrimes;
	int j=0;
	while(1)
	{
		j=primes[--gdzie];
		//printf("%d ",j);
		if ((TARGET%j==0) && isPrime(TARGET/j))
		{
			printf("wynik %d ",j);
			break;
		}
	}

	free(primes);
	return 0;
}


	/* printf("%d ",biggestPossible);
	if (isPrime(17))
		puts("17 prime");
	if (isPrime(80))
		puts("80 prime");
	else
		puts("80 complex"); */



